src/posix/platform/udp.cpp

```diff
@@ -152,8 +152,16 @@ static otError transmitPacket(int aFd, uint8_t *aPayload, uint16_t aLength, cons
 #else
     msg.msg_controllen = controlLength;
 #endif
-
-    rval = sendmsg(aFd, &msg, 0);
+    for (size_t i = 0; i < OPENTHREAD_CONFIG_PLATFORM_UDP_SEND_RETRY; i++)
+    {
+        rval = sendmsg(aFd, &msg, 0);
+        if (rval > 0)
+        {
+            break;
+        }
+        otLogWarnPlat("sendmsg failed, retry: %s", strerror(errno));
+        sleep(0);
```

Comment:
Is this necessary? Why would it make difference to retry multiple times continuously?

Reply:
Looks like the kernel becomes unavailable for sending for some time. So we yield and retry.

Reply:
But why `sleep(0)`? I am also curious why would it gives `EINVAL`?

Reply:
I think this is caused by adding address is not taking effect immediatelly.